### PR TITLE
Fix display of non-numeric DataFrame index columns

### PR DESCRIPTION
**Pull request by joyceerhl**
For #5253 

The change here is to show the index column in addition to the row number column only for pandas DataFrames and Series objects, because index columns are meaningful for DFs and Series but for ndarrays, lists, dicts and tensors the index column is generated by us as part of the DataFrame conversion code and is hence not meaningful to the user.

DataFrame:
image

Same data as an ndarray (note that there isn't actually an index column being generated on the ndarray itself):
image

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py [hunk 1]
```diff
@@ -170,6 +170,17 @@ def _VSCODE_getDataFrameInfo(df):
 
     columnTypes = _VSCODE_builtins.list(df.dtypes)
 
+    # Compute the index column. It may have been renamed
+    try:
```
Comment (IanMatthewHuff) [inline pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:174]: This looks a bit different from the code in getJupyterVariableDataFrameInfo.py that does something similar with index column. Looks like your version might be safer with the try / except. Should that maybe be updated as well?

Comment (joyceerhl) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:174]: This is actually just restoring code that I removed several PRs ago. Rich pointed out at the time that it was required and now I see why 😊

Comment (IanMatthewHuff) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:174]: Sorry, maybe I didn't say that right. I was saying that this code here looked safer than the code in the other location. Just wondering if getJupyterVariableDataFrameInfo.py should be upgraded with the try / except that you have here.

Comment (joyceerhl) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:174]: Ah good point, will update

**Pull request merged**